### PR TITLE
[INT-303] - Switch to pydantic 2.

### PR DESCRIPTION
**Pull request by polina-koval**
https://huntflow.atlassian.net/browse/INT-303

### CODE REVIEW

File: huntflow_api_client/models/common.py [hunk 1]
```diff
@@ -23,58 +23,52 @@ def jsonable_dict(
...
-            "skip_defaults": skip_defaults,
             "exclude_unset": exclude_unset,
             "exclude_defaults": exclude_defaults,
             "exclude_none": exclude_none,
-            "encoder": encoder,
+            "round_trip": round_trip,
+            "warnings": warnings,
         }
-        return json.loads(self.json(**params, **dumps_kwargs))  # type: ignore
+        return json.loads(self.model_dump_json(**params))  # type: ignore
```
Comment (aleks-v-k) [inline huntflow_api_client/models/common.py:42]: You have all possible parameters, so it's not needed to build a dict, just pass all the params to the model_dump_json:
```
model_dump_json(inlude=include, exclude=exclude, ...)
```

File: huntflow_api_client/models/request/resume.py [hunk 2]
```diff
@@ -6,17 +6,17 @@
...
 class ApplicantResumeUpdateData(BaseModel):
-    body: Optional[str] = Field(..., description="Resume text")
+    body: Optional[str] = Field(None, description="Resume text")
 
 
 class ApplicantResumeUpdateRequest(JsonRequestModel):
-    account_source: Optional[PositiveInt] = Field(..., description="Resume source ID")
+    account_source: Optional[PositiveInt] = Field(None, description="Resume source ID")
     data: Optional[ApplicantResumeUpdateData] = Field(
-        ...,
```
Comment (aleks-v-k) [inline huntflow_api_client/models/request/resume.py:15]: Do you intentionally change the field's signature? `...` -> `None`

File: pyproject.toml [hunk 3]
```diff
@@ -10,6 +10,7 @@ dependencies = [
     "httpx>=0.23.3",
     "email-validator>=1.3.1",
     "pydantic>=1.10.4",
```
Comment (aleks-v-k) [inline pyproject.toml:12]: I thought you want to use pydantic 2.*

File: huntflow_api_client/models/request/resume.py (hunk 2 above)
Comment (polina-koval) [reply huntflow_api_client/models/request/resume.py:15]: No 😢  Looks like bump-pydantic tricks.

File: huntflow_api_client/models/common.py [hunk 4]
```diff
@@ -11,35 +12,34 @@
     VacancyState,
 )
 
-IntStr = Union[int, str]
-AbstractSetIntStr = AbstractSet[IntStr]
-MappingIntStrAny = Mapping[IntStr, Any]
+IncEx: typing_extensions.TypeAlias = "Set[int] | Set[str] | Dict[int, Any] | Dict[str, Any] | None"
```
Comment (aleks-v-k) [inline huntflow_api_client/models/common.py:15]:
```
_FieldSet: typing_extensions.TypeAlias = "Set[int] | Set[str] | Dict[int, Any] | Dict[str, Any]"
...
include: Optional[_FieldSet] = None
...
```

**Review by aleks-v-k — APPROVED** [review]

**Pull request merged**